tardis/tests/integration_tests/conftest.py

```diff
@@ -77,15 +77,18 @@ def plot_object(request):
 def data_path(request):
     integration_tests_config = request.config.integration_tests_config
     hdf_filename = "{0}.h5".format(os.path.basename(request.param))
+    if (request.config.getoption("--generate-reference") ):
+        ref_path = os.path.join(os.path.expandvars(os.path.expanduser(
+            integration_tests_config['reference'])), tardis_githash[:7]
+        )
+    else:
+        ref_path = os.path.join(os.path.expandvars(
+            os.path.expanduser(integration_tests_config['reference'])), hdf_filename
+        )
 
     path = {
         'config_dirpath': request.param,
-        'reference_filepath': os.path.join(os.path.expandvars(
-            os.path.expanduser(integration_tests_config['reference'])), hdf_filename
-        ),
-        'gen_ref_path': os.path.join(os.path.expandvars(os.path.expanduser(
-            integration_tests_config['generate_reference'])), tardis_githash[:7]
-        ),
+        'reference_filepath': ref_path,
```

Comment:
You might want to think of another name here, because when you generate references, it would be like:
`path/as/mentioned/in/yml/githash` which is not "filepath" per se.